test/test_dos.py

```diff
@@ -5457,18 +5456,21 @@ def _test_cpu(self, cpu_vm, cpu_vm_dpmi, cpu_emu):
 $_ignore_djgpp_null_derefs = (off)
 """%(cpu_vm, cpu_vm_dpmi, cpu_emu))
 
-        with open("test-imagedir/dXXXXs/c/test.log") as f:
-            lines = list(f)
-            self.assertEqual(len(lines), 5056)
-            # compare or copy to reference file
-            try:
-                with open("test-i386.log") as g:
-                    self.maxDiff = None
-                    self.assertEqual(list(g), lines)
-            except IOError:
-                # copy as reference file
-                with open("test-i386.log", "w") as g:
-                    g.write("".join(lines))
+        try:
```

Comment:
The only possible way to check for
an open failure? (I am ignorant about
python)

Reply:
With python generally exceptions are raised when things don't work. You could check if the file exists, but it's a race and won't catch other things like unreadable file etc.

Reply:
Yes that's the normal way in Python to check errors. If you don't the Python interpreter will terminate with a message about the file and a backtrace.
```
$ python
Python 3.7.5 (default, Dec 10 2019, 19:52:22) 
[GCC 9.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> open('notexist.txt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'notexist.txt'
```